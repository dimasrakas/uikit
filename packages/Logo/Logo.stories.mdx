import { Meta, Story, Canvas } from "@storybook/addon-docs/blocks";
import { ArgsTable } from "@storybook/addon-docs";

import Logo from "./Logo.vue";

<Meta
  title="Design System/Logo"
  component={Logo}
  argTypes={{
    variant: {
      control: "select",
      options: [
        "primary",
        "black",
        "white",
        "icon-primary",
        "icon-black",
        "icon-white",
      ],
    },
  }}
/>

export const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { Logo },
  template: '<Logo v-bind="$props" />',
});

# Logo

With `MDX`, we can define a story for `Logo` right in the middle of our
Markdown documentation.

## Install

```bash
yarn add @dimasrakas/Logo

// or

npm install @dimasrakas/Logo
```

## Imports

```js
import { Logo } from "@dimasrakas/logo";
```

## Usage

### Primary

<Canvas withSource="open">
  <Story name="Primary" args={{ variant: "primary", height: 128 }}>
    {Template.bind({})}
  </Story>
</Canvas>

### Black

<Canvas withSource="open">
  <Story name="Black" args={{ variant: "black", height: 128 }}>
    {Template.bind({})}
  </Story>
</Canvas>

### White

<Canvas withSource="open">
  <Story name="White" args={{ variant: "white", height: 128 }}>
    {Template.bind({})}
  </Story>
</Canvas>

### Icon Primary

<Canvas withSource="open">
  <Story name="Icon Primary" args={{ variant: "icon-primary", height: 180 }}>
    {Template.bind({})}
  </Story>
</Canvas>

### Icon Black

<Canvas withSource="open">
  <Story name="Icon Black" args={{ variant: "icon-black", height: 180 }}>
    {Template.bind({})}
  </Story>
</Canvas>

### Icon White

<Canvas withSource="open">
  <Story name="Icon White" args={{ variant: "icon-white", height: 180 }}>
    {Template.bind({})}
  </Story>
</Canvas>

## Props

<ArgsTable of={Logo} />
