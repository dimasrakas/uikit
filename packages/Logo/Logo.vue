<template>
  <img :height="height" width="auto" :src="logo" alt="" />
</template>

<script>
export default {
  name: "Logo",
  props: {
    variant: {
      type: String,
      default: "primary",
    },
    height: {
      type: Number,
      default: 40,
    },
  },
  computed: {
    logo() {
      if (this.variant === "primary") {
        return require("./assets/primary.svg");
      } else if (this.variant === "black") {
        return require("./assets/black.svg");
      } else if (this.variant === "white") {
        return require("./assets/white.svg");
      } else if (this.variant === "icon-primary") {
        return require("./assets/icon-primary.svg");
      } else if (this.variant === "icon-black") {
        return require("./assets/icon-black.svg");
      } else if (this.variant === "icon-white") {
        return require("./assets/icon-white.svg");
      } else {
        return require("./assets/primary.svg");
      }
    },
  },
};
</script>
