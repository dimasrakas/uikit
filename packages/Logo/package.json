{
  "name": "@dimasrakas/logo",
  "version": "0.0.1",
  "publishConfig": {
    "access": "public"
  }
}
