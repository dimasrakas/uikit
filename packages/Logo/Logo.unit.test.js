import { shallowMount } from "@vue/test-utils";
import Logo from "./Logo.vue";

describe("Button.vue", () => {
  it('should have "Logo" as name', () => {
    expect(Logo.name).toBe("Logo");
  });

  it("should render correctly", () => {
    const wrapper = shallowMount(Logo);
    expect(wrapper.element).toMatchSnapshot();
  });
});
