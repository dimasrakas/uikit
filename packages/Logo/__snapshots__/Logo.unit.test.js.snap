// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue should render correctly 1`] = `
<img
  alt=""
  height="40"
  src=""
  width="auto"
/>
`;
